### PR TITLE
Add an "override" to the usePropertyIfPresent to return a `ResolverResult`

### LINKED ISSUE #37: Something to consider...should `usePropertyIfPresent` return a value? Because it's a bit unfortunate that you have to create a mutable map and mutate it in each lambda here.
Something to consider...should `usePropertyIfPresent` return a value? Because it's a bit unfortunate that you have to create a mutable map and mutate it in each lambda here.

I don't think this should be done in this PR, but maybe consider whether we should add a new issue to change that.

_Originally posted by @sleberknight in https://github.com/kiwiproject/dropwizard-config-providers/pull/36#discussion_r494289773_

### PR DESCRIPTION
**Pull request by chrisrohr**
Fixes #37

### CODE REVIEW

File: src/main/java/org/kiwiproject/config/provider/util/SinglePropertyResolver.java [hunk 1]
```diff
@@ -82,7 +78,7 @@
      * @param <T>                    The type of the value to be returned
      * @return The resolved value
      */
-    @SuppressWarnings({"unchecked", "java:S107"})
+    @SuppressWarnings({"java:S107"})
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/config/provider/util/SinglePropertyResolver.java:81]: At this number of parameters...is it worth doing an **[Extract Parameter Object](https://www.jetbrains.com/help/idea/extract-into-class-refactorings.html#extract_parameter_object)** refactoring?

Comment (sleberknight) [reply src/main/java/org/kiwiproject/config/provider/util/SinglePropertyResolver.java:81]: ...and adding a Lombok `@Builder` to the extracted parameter object.

**Review by sleberknight — CHANGES_REQUESTED** [review]
See my comments

File: src/main/java/org/kiwiproject/config/provider/util/SinglePropertyResolver.java (hunk 1 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/config/provider/util/SinglePropertyResolver.java:81]: I didn't really want to, but I can

**Review by sleberknight — APPROVED** [review]

**Pull request merged**